crates/lib3h/src/error.rs

```diff
@@ -73,6 +73,23 @@ pub enum ErrorKind {
     __Nonexhaustive,
 }
 
+impl Clone for ErrorKind {
+    fn clone(&self) -> Self {
+        match self {
+            ErrorKind::GhostError(e) => ErrorKind::GhostError(e.clone()),
+            ErrorKind::Io(e) => ErrorKind::Io(e.kind().into()),
+            ErrorKind::TransportError(e) => ErrorKind::TransportError(e.clone()),
+            ErrorKind::Lib3hProtocolError(e) => ErrorKind::Lib3hProtocolError(e.clone()),
+            ErrorKind::HcId(e) => ErrorKind::HcId(e.clone()),
+            ErrorKind::RmpSerdeDecodeError(e) => ErrorKind::Other(format!("{:?}", e)),
```

Comment:
This is really the only questionable part... but the plan is to move away from RmpSerde so it shouldn't be a problem in the future, and in the mean time we still at least get the error data.